src/snowflake/snowpark/dataframe.py

```diff
@@ -3925,7 +3950,7 @@ def write(self, _emit_ast: bool = True) -> DataFrameWriter:
         """
 
         # AST.
-        if _emit_ast and self._ast_id is not None:
+        if self._ast_id is not None:
```

Comment:
Oh, I see. This is a bit of a hack, but probably reasonable given the alternative.

Reply:
Yeah, since `write` is a property, I can't use the `_emit_ast` parameter.